applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -0,0 +1,303 @@
+        }
+
+        return result;
+    }
+
+    BoundedMatrix<double, TNumNodes, TNumNodes> CalculateConductivityMatrix(const GeometryType::ShapeFunctionsGradientsType& rShapeFunctionGradients,
+                                                                            const Vector& rIntegrationCoefficients,
+                                                                            const ProcessInfo& rCurrentProcessInfo) const
+    {
+        GeoThermalDispersionLaw law{TDim};
+        const auto constitutive_matrix =
+                law.CalculateThermalDispersionMatrix(GetProperties(), rCurrentProcessInfo, GetGeometry());
+
+        auto result = BoundedMatrix<double, TNumNodes, TNumNodes>{ZeroMatrix{TNumNodes, TNumNodes}};
+        for (unsigned int GPoint = 0; GPoint < GetGeometry().IntegrationPointsNumber(GetIntegrationMethod()); ++GPoint) {
+            BoundedMatrix<double, TDim, TNumNodes> Temp = prod(constitutive_matrix, trans(rShapeFunctionGradients[GPoint]));
+            result += prod(rShapeFunctionGradients[GPoint], Temp) * rIntegrationCoefficients[GPoint];
+        }
+
+        return result;
+    }
+
+    BoundedMatrix<double, TNumNodes, TNumNodes> CalculateCapacityMatrix(const Vector& rIntegrationCoefficients) const
+    {
+        const auto& r_properties = GetProperties();
+        const auto  cWater = r_properties[POROSITY] * r_properties[SATURATION] *
+                             r_properties[DENSITY_WATER] * r_properties[SPECIFIC_HEAT_CAPACITY_WATER];
+        const auto  cSolid = (1.0 - r_properties[POROSITY]) *
+                             r_properties[DENSITY_SOLID] * r_properties[SPECIFIC_HEAT_CAPACITY_SOLID];
+
+        auto result = BoundedMatrix<double, TNumNodes, TNumNodes>{ZeroMatrix{TNumNodes, TNumNodes}};
+        const auto& NContainer = GetGeometry().ShapeFunctionsValues(GetIntegrationMethod());
+        for (unsigned int GPoint = 0; GPoint < GetGeometry().IntegrationPointsNumber(GetIntegrationMethod()); ++GPoint) {
+            const auto N = Vector{row(NContainer, GPoint)};
+            result += (cWater + cSolid) * outer_prod(N, N) * rIntegrationCoefficients[GPoint];
+        }
+
+        return result;
+    }
+
+    array_1d<double, TNumNodes> GetNodalValuesOf(const Variable<double>& rNodalVariable) const
+    {
+        auto result = array_1d<double, TNumNodes>{};
+        const auto& r_geometry = GetGeometry();
+        std::transform(r_geometry.begin(), r_geometry.end(), result.begin(),
+                       [&rNodalVariable](const auto& node) {
+                           return node.FastGetSolutionStepValue(rNodalVariable);
+                       });
+        return result;
+    }
+
+    void CheckProperty(const Kratos::Variable<double>& rVariable) const
+    {
+        KRATOS_ERROR_IF_NOT(GetProperties().Has(rVariable)) << rVariable.Name()
+                                                            << " does not exist in the material properties" << std::endl;
```

Comment:
is 'material properties' in this case the right wording, maybe 'thermal element' is more descriptive?